boxoffice/views/admin_discount.py

```diff
@@ -0,0 +1,210 @@
+            discount_policies=discount_policies,
+            total_pages=total_pages,
+            paginated=(total_policies > results_per_page),
+            current_page=page)
+    else:
+        return dict(title=organization.title)
+
+
+@app.route('/admin/o/<org>/discount_policy/new', methods=['OPTIONS', 'POST'])
+@lastuser.requires_login
+@load_models(
+    (Organization, {'name': 'org'}, 'organization'),
+    permission='org_admin'
+    )
+@xhr_only
+def admin_add_discount_policy(organization):
+    discount_policy_form = DiscountPolicyForm(request.form)
+    all_items = Item.query.join(ItemCollection).filter(
+        ItemCollection.organization == organization).options(db.load_only('id')).all()
+    discount_policy_form.items.query = all_items
+    if not discount_policy_form.validate_on_submit():
+        return api_error(message=discount_policy_form.errors, status_code=400)
+
+    discount_policy = DiscountPolicy(organization=organization)
+    if discount_policy_form.discount_type.data:
+        discount_policy.set_secret()
+    if not discount_policy_form.is_price_based.data:
+        discount_policy.items.extend(discount_policy_form.items.data)
+    else:
+        discount_price_form = DiscountPriceForm(request.form)
+        discount_price_form.item.query = all_items
+        if not discount_price_form.validate_on_submit():
+            return api_error(message=discount_price_form.errors, status_code=400)
+    discount_policy_form.populate_obj(discount_policy)
+    discount_policy.make_name()
+    db.session.add(discount_policy)
+    db.session.commit()
+
+    if discount_policy.is_price_based:
+        discount_price = Price(discount_policy=discount_policy)
+        discount_price_form.populate_obj(discount_price)
+        discount_price.make_name()
+        db.session.add(discount_price)
+        discount_policy.items.append(discount_price.item)
+        db.session.commit()
+    return api_success(result={'discount_policy': jsonify_discount_policy(discount_policy)},
+        doc="New discount policy created.", status_code=200)
+
+
+@app.route('/admin/o/<org>/discount_policy/<discount_policy_id>/edit', methods=['OPTIONS', 'POST'])
+@lastuser.requires_login
+@load_models(
+    (Organization, {'name': 'org'}, 'organization'),
+    (DiscountPolicy, {'id': 'discount_policy_id'}, 'discount_policy'),
+    permission='org_admin'
+    )
+@xhr_only
+def admin_edit_discount_policy(organization, discount_policy):
+    discount_policy_form = DiscountPolicyForm(request.form)
+    all_items = Item.query.join(ItemCollection).filter(
+        ItemCollection.organization == organization).options(db.load_only('id')).all()
+    discount_policy_form.items.query = all_items
+
+    if not discount_policy_form.validate_on_submit():
+        return api_error(message=discount_policy_form.errors, status_code=400)
+
+    if not discount_policy.is_price_based:
+        discount_policy.items.extend(discount_policy_form.items.data)
```

Comment:
Shouldn't this replace rather than extend? How do you remove the items it's associated with?